### PR TITLE
Vam 70 discontinue reasons

### PR DESCRIPTION
**Pull request by sidjtd**
4 files!

### CODE REVIEW

File: services/orderMetaService/discontinueReasons.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+                        inactive: { $ne: 'INACTIVE' },
+                        localStationNumber: stnId,
+                    }, { id: { $in: paramCandidates } }],
+                },
+                projection: serviceUtils.createProjection(['id', 'label']),
+            }).then((res) => {
+                callback(null, res);
+            }).catch(err => callback(err));
+        }).catch(err => callback(err));
+    }).catch(err => callback(err));
```
Comment (mfuroyama) [inline services/orderMetaService/discontinueReasons.js]: It looks like this works, but it's **ever** so slightly hard to follow. This is okay for now, I think, but after build release 1, let's consider revisiting this implementation, and maybe cleaning this up (e.g. let's think about ways to  modularize the functional components of this function, or perhaps we could involve the `async` library to make this a little bit more readable)

File: services/orderMetaService/discontinueReasons.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * @summary RPC to return a list of valid discontinuation reasons.
+ * @param  {Object}   context  Session context information associated with the service call.
+ * @param  {Function} callback Function called on process completion, with standard (err, result) signature.
+ * @return {Array}             Array of labels and urn id's
+ *
+ * @memberof OrderMetaService
+ * @instance
+ */
+function discontinueReasons(context, callback) {
```
Comment (mfuroyama) [inline services/orderMetaService/discontinueReasons.js]: Ahh....let's rename this to something like `lookupDiscontinueReasons`...sorry... :/

Comment (sidjtd) [reply services/orderMetaService/discontinueReasons.js]: A split second b4 I read this comment I saw the need of this fix,  thought to myself "Nice!! I'm ahead of this game, baby!" but then I saw you posted this 18 hours ago... 

But yes, will change it!

Comment (mfuroyama) [reply services/orderMetaService/discontinueReasons.js]: Actually, after talking to Conor, let's rename this function to `listDiscontinueReasons`

Check out the additions to commandment #4 in the [Holy VAM Commandments](https://github.com/vistadataproject/VICSServer/wiki/Guidelines-for-VICS-Services)

On a side note, this is one of those things we'll need to revisit going forward with refactoring.

Comment (sidjtd) [reply services/orderMetaService/discontinueReasons.js]: roger!

File: services/orderMetaService/listDiscontinueReasons.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * @summary RPC to return a list of valid discontinuation reasons.
+ * @param  {Object}   context  Session context information associated with the service call.
+ * @param  {Function} callback Function called on process completion, with standard (err, result) signature.
+ * @return {Array}             Array of labels and urn id's
+ *
+ * @memberof OrderMetaService
+ * @instance
+ */
+function discontinueReasons(context, callback) {
```
Comment (mfuroyama) [inline services/orderMetaService/listDiscontinueReasons.js]: I think my original comments got lost somehow:

Let's change the service function name to `listDiscontinueReasons`. This is per Conor's rule #4 in the VAM Commandments (prefix `lookup` when getting 1 or more things, `list` when getting all the things...)

Comment (sidjtd) [reply services/orderMetaService/listDiscontinueReasons.js]: Noooo it didn't get lost, I just forgot to commit and push those changes!!!!

File: services/tests/discontinueReasons-spec.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+describe('discontinueReasons asynchronous specs', () => {
+    let context;
+    beforeAll(() => {
+        context = serviceTestUtils.beforeAllScript(modelFunction).context;
+    });
+
+
+    it('Function returns array of objects with id and label key', (done) => {
+        modelFunction.discontinueReasons(context, (err, res) => {
```
Comment (mfuroyama) [inline services/tests/discontinueReasons-spec.js]: I'm guessing this line will probably raise an error in the service test spec runs... :)

File: services/tests/discontinueReasons-spec.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+            if (err) console.log(err);
+            res.forEach((el) => {
+                expect(_.has(el, 'id')).toEqual(true);
```
Comment (mfuroyama) [inline services/tests/discontinueReasons-spec.js]: This is okay, but you might also want to consider using `toBeDefined` here, so
```Javascript
expect(el).toBeDefined();
expect(el.id).toBeDefined();
``` etc.

File: services/tests/discontinueReasons-spec.js (hunk 4 above)
Comment (sidjtd) [reply services/tests/discontinueReasons-spec.js]: Man, my mind must be floating :/:/:/:/ My bad!!!!!!

Comment (sidjtd) [reply services/tests/discontinueReasons-spec.js]: next time, find > replaceAll.... then eye ball.... why didnt I do that....
